src/_background_loader.1.ts

```diff
@@ -13,6 +13,7 @@ import * as PluginService from './extension/background-script/PluginService'
 import * as HelperService from './extension/background-script/HelperService'
 import * as NonceService from './extension/background-script/NonceService'
 import * as ProviderService from './extension/background-script/ProviderService'
+import { upload as pluginArweaveUpload } from './plugins/FileService/arweave/index'
```

Comment:
Why not use plugin service?
You can define at `src/plugins/Wallet/wallet.ts` and invoke it like `Services.Plugin.invokePlugin('maskbook.file', 'method_name')`

Reply:
the is generator
from @Jack-Works

Reply:
it's not a normal async function. it's a async generator so it have no infra for this